bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImpl.java

```diff
@@ -201,10 +222,25 @@ private Integer parseCategoryId() {
 
         try {
             categoryId = Integer.parseInt(this.request.getRequestPathInfo().getSelectorString());
-        } catch (NumberFormatException nef) {
+        } catch (NullPointerException | NumberFormatException nef) {
             LOGGER.warn("Could not parse category id from current page selectors.");
         }
         return categoryId;
     }
+
+    @Override
+    public int getTotalCount() {
```

Comment:
Move these three methods up closer to the `getTitle` methods to keep the file structured.